### PR TITLE
fix #23465 - insert measure before / after barlines (& clefs & signatures)

### LINKED ISSUE #23465: Insert measure after selection (selected barline) puts new measure(s) before barlines of the current measure instead of after
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

_No response_

### Steps to reproduce

1. create a score
2. put a repeat or double barline at the end of the last measure (or anywhere in the score)
3. select the measure that ends with a repeat or double barline
4. in the menu choose Add > Measures > Insert after selection
5. enter any number of measures and hit OK
The inserted measures will be **before** the double barlines/repeats instead of after it. The same problem occurs whether you select the measure itself or the barlines, there is no way to insert after the barline or repeat. Either both cases should insert after the barlines, or if you select the measure then the insert could insert after the current measure but before the barlines, and if you select the barlines themselves then the insert could insert the new measure after the barline. There should be some way to do so.

Expected result: [see comment](https://github.com/musescore/MuseScore/issues/23465#issuecomment-2253090805) and [comment ](https://github.com/musescore/MuseScore/issues/23465#issuecomment-2253435392)

### Screenshots/Screen recordings

_No response_

### MuseScore Version

MuseScore Studio version (64-bit): 4.3.2-241630831, revision: github-musescore-musescore-22b46f2

### Regression

I don't know

### Operating system

OS: Windows 10 Version 2009 or later, Arch.: x86_64, 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sammik**
Resolves: #23465 
Resolves: #23676

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

File: src/notation/internal/notationinteraction.cpp [hunk 1]
```diff
@@ -3779,10 +3782,37 @@ void NotationInteraction::addBoxes(BoxType boxType, int count, AddBoxesTarget ta
...
+            ElementType selectedItemType = selectedItem->type();
+            if (selectedItemType == ElementType::CLEF || selectedItemType == ElementType::BAR_LINE
+                || selectedItemType == ElementType::TIMESIG || selectedItemType == ElementType::KEYSIG) {
+                Fraction itemTick = selectedItem->tick();
+                Fraction measureTick = selectedItemMeasure->tick();
+                Fraction measureLastTick = measureTick + selectedItemMeasure->ticks();
+
+                if (itemTick == measureTick) {
+                    if (target == AddBoxesTarget::AfterSelection) {
+                        beforeBoxIndex -= 1;
```
Comment (cbjeukendrup) [inline src/notation/internal/notationinteraction.cpp:3808]: I found one odd case. When you create a clef and use the Properties panel to put it after the barline,
Scherm­afbeelding 2024-07-18 om 20 08 36
and then insert a measure after it, the measure goes before it:
Scherm­afbeelding 2024-07-18 om 20 11 22
Perhaps the problem is that the clef should not be moved in this case.

Comment (sammik) [reply src/notation/internal/notationinteraction.cpp:3808]: Thank You, Ill take a look.

Comment (sammik) [reply src/notation/internal/notationinteraction.cpp:3808]: Ok, I think #23676 is an issue (I reported it separately, as I think, it doesnt belong here), will try to fix it too.

**Comment by sammik** [comment]
@cbjeukendrup it should be fixed now

I needed to rewrite it a bit, but I commented code, to be more clear on first view

**Comment by sammik** [comment]
vtests probable needed rebase

**Review by cbjeukendrup — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
Tested #23465 #23676 on Win10, Mac13.6 - **FIXED**

**Pull request merged**